### PR TITLE
Update `swc_core` to `v0.89.x`

### LINKED ISSUE #56144: Wrong import with prisma client and nextJest config
### Link to the code that reproduces this issue

https://codesandbox.io/p/sandbox/thirsty-joji-tgpmz4

### To Reproduce

1. Install dependencies
2. Run `yarn test`

### Current vs. Expected behavior

Fails with the following error, reporting an import from `./client/client`, while the source file only has `./client`;
image

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 23.0.0: Fri Sep 15 14:41:34 PDT 2023; root:xnu-10002.1.13~1/RELEASE_ARM64_T8103
    Binaries:
      Node: 18.13.0
      npm: 8.19.3
      Yarn: 3.6.0
      pnpm: N/A
    Relevant Packages:
      next: 13.5.2
      eslint-config-next: 13.5.2
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: standalone
```

### Which area(s) are affected? (Select all that apply)

Jest (next/jest), SWC transpilation

### Additional context
- Same setup was working before I upgraded next from 13.1.6
- Overriding the transform in jest config fixes the issue

<sub>[WEB-1682](https://linear.app/vercel/issue/WEB-1682/wrong-import-with-prisma-client-and-nextjest-config)</sub>

### LINKED ISSUE #59805: Incorrect CSS string escaping for `content` property with styled-components
### Link to the code that reproduces this issue

https://github.com/yellcorp/next14-styco-escaping-bug

### To Reproduce

1. Start the application in development (`npm run dev`)
2. Access the `/` path in a web browser
3. Observe the test message

### Current vs. Expected behavior

Given the component's source, I would expect the following to be displayed:

> ©This should be preceded by a copyright symbol and _not_ backslash, letter A, digit nine

Instead, the following is displayed:

> \A9This should be preceded by a copyright symbol and _not_ backslash, letter A, digit nine

The source code has the `content` property set to two backslashes followed by `A9`. I would expect the interpretation of the JS TTL to produce a single backslash followed by `A9`, and from there, interpretation as CSS should produce a ©. Instead it seems as if the TTL content is being forwarded as CSS verbatim. The same behavior is seen with any valid hex-escaped character. 

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 22.6.0: Wed Oct  4 21:25:40 PDT 2023; root:xnu-8796.141.3.701.17~4/RELEASE_ARM64_T8103
Binaries:
  Node: 18.18.2
  npm: 9.8.1
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.0.5-canary.19
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.1.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

SWC transpilation

### Additional context

The last release of Next to pass this test was `14.0.4-canary.18`.
The first release that fails is `14.0.4-canary.22`.
I see there are canary _tags_ .19 through .21, which were not released. I wasn't able to test them but they may also be suspect.

The title mentions the `content` property but this is likely an issue with any CSS property that accepts a string. I suspect the problem is with some Next.js dependency, but I'm not familiar enough with Next internals to route this report further.

The repro repo also has a number of branches demonstrating some workarounds in case it helps narrow down the cause. Any one of the following actions produces the expected result.
- remove `compiler.styledComponents` option
- alias the TTL function to some other name and use that instead
- avoid escapes and use the literal character (obviously! but my use case is to make non-breaking spaces obvious)
- surround the TTL name in parens and place the opening backtick after the closing paren
- use `next dev --turbo`

### PR DESCRIPTION
**Pull request by kdy1**
## Turbopack

* https://github.com/vercel/turbo/pull/7092 

### What?

Update SWC crates

### Why?

To fix some issues

### How?

Fixes #59805
Fixes #56144

Closes PACK-2222
Closes PACK-2285

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | vercel/next.js kdy1/swc-core-89-1 | Change |
| - | - | - | - |
| buildDuration | 11.9s | 11.9s | N/A |
| buildDurationCached | 6.1s | 5s | N/A |
| nodeModulesSize | 200 MB | 200 MB | N/A |
| nextStartRea..uration (ms) | 409ms | 407ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js kdy1/swc-core-89-1 | Change |
| - | - | - | - |
| 3f784ff6-HASH.js gzip | 53.4 kB | 53.4 kB | ✓ |
| 423.HASH.js gzip | 185 B | 181 B | N/A |
| 68-HASH.js gzip | 29.8 kB | 29.8 kB | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 238 B | 238 B | ✓ |
| main-HASH.js gzip | 31.8 kB | 31.8 kB | N/A |
| webpack-HASH.js gzip | 1.7 kB | 1.7 kB | ✓ |
| Overall change | 101 kB | 101 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js kdy1/swc-core-89-1 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js kdy1/swc-core-89-1 | Change |
| - | - | - | - |
| _app-HASH.js gzip | 194 B | 195 B | N/A |
| _error-HASH.js gzip | 182 B | 181 B | N/A |
| amp-HASH.js gzip | 502 B | 502 B | ✓ |
| css-HASH.js gzip | 320 B | 322 B | N/A |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | N/A |
| edge-ssr-HASH.js gzip | 255 B | 256 B | N/A |
| head-HASH.js gzip | 350 B | 349 B | N/A |
| hooks-HASH.js gzip | 368 B | 369 B | N/A |
| image-HASH.js gzip | 4.18 kB | 4.18 kB | N/A |
| index-HASH.js gzip | 257 B | 256 B | N/A |
| link-HASH.js gzip | 2.61 kB | 2.61 kB | N/A |
| routerDirect..HASH.js gzip | 310 B | 311 B | N/A |
| script-HASH.js gzip | 384 B | 383 B | N/A |
| withRouter-HASH.js gzip | 306 B | 308 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 608 B | 608 B | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js kdy1/swc-core-89-1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 484 B | 484 B | ✓ |
| Overall change | 484 B | 484 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js kdy1/swc-core-89-1 | Change |
| - | - | - | - |
| index.html gzip | 527 B | 529 B | N/A |
| link.html gzip | 539 B | 541 B | N/A |
| withRouter.html gzip | 521 B | 524 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js kdy1/swc-core-89-1 | Change |
| - | - | - | - |
| edge-ssr.js gzip | 94 kB | 94 kB | N/A |
| page.js gzip | 149 kB | 149 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js kdy1/swc-core-89-1 | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 625 B | 622 B | N/A |
| middleware-r..fest.js gzip | 151 B | 149 B | N/A |
| middleware.js gzip | 37.6 kB | 37.6 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 1.92 kB | 1.92 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js kdy1/swc-core-89-1 | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 170 kB | 170 kB | ✓ |
| app-page-exp..prod.js gzip | 95.7 kB | 95.7 kB | ✓ |
| app-page-tur..prod.js gzip | 96.3 kB | 96.3 kB | ✓ |
| app-page-tur..prod.js gzip | 90.9 kB | 90.9 kB | ✓ |
| app-page.run...dev.js gzip | 142 kB | 142 kB | ✓ |
| app-page.run..prod.js gzip | 90.2 kB | 90.2 kB | ✓ |
| app-route-ex...dev.js gzip | 22.2 kB | 22.2 kB | ✓ |
| app-route-ex..prod.js gzip | 14.9 kB | 14.9 kB | ✓ |
| app-route-tu..prod.js gzip | 14.9 kB | 14.9 kB | ✓ |
| app-route-tu..prod.js gzip | 14.5 kB | 14.5 kB | ✓ |
| app-route.ru...dev.js gzip | 21.6 kB | 21.6 kB | ✓ |
| app-route.ru..prod.js gzip | 14.5 kB | 14.5 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.42 kB | 9.42 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.69 kB | 9.69 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.42 kB | 9.42 kB | ✓ |
| pages-turbo...prod.js gzip | 22 kB | 22 kB | ✓ |
| pages.runtim...dev.js gzip | 22.6 kB | 22.6 kB | ✓ |
| pages.runtim..prod.js gzip | 22 kB | 22 kB | ✓ |
| server.runti..prod.js gzip | 49.7 kB | 49.7 kB | ✓ |
| Overall change | 932 kB | 932 kB | ✓ |

</details>

</details>
Commit: 4eaa68eb493fc56306d7a765063161390964eb78

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 4eaa68eb493fc56306d7a765063161390964eb78

`TURBOPACK=1 pnpm test-dev test/e2e/app-dir/emotion-js/index.test.ts` (turbopack)
- app dir - emotion-js > should render emotion-js css with compiler.emotion option correctly

<details>
<summary>Expand output</summary>

  ● app dir - emotion-js › should render emotion-js css with compiler.emotion option correctly

    page.waitForSelector: Timeout 60000ms exceeded.
    =========================== logs ===========================
    waiting for locator('h1')
    ============================================================

      422 |     return this.chain(() => {
      423 |       return page
    > 424 |         .waitForSelector(selector, { timeout, state: 'attached' })
          |          ^
      425 |         .then(async (el) => {
      426 |           // it seems selenium waits longer and tests rely on this behavior
      427 |           // so we wait for the load event fire before returning

      at waitForSelector (lib/browsers/playwright.ts:424:10)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by kdy1** [comment]
Regression: https://github.com/swc-project/swc/compare/6d9939d18eb645179c2436847fa0e711a89ff664...1890e66023600564a531e81a99dc7db78772951d

**Comment by kdy1** [comment]
## Investigation

https://github.com/vercel/next.js/commit/4eaa68eb493fc56306d7a765063161390964eb78 + https://github.com/vercel/turbo/commit/f63cc49ada32a2d7788a48eb53c3ab3768b69cdd fails

**Comment by kdy1** [comment]
https://github.com/swc-project/swc/pull/8530 was the cause of the regression

File: packages/next-swc/crates/next-custom-transforms/src/transforms/react_server_components.rs [hunk 1]
```diff
@@ -642,7 +642,10 @@ pub fn server_components<C: Comments>(
     as_folder(ReactServerComponents {
         is_react_server_layer,
         comments,
-        filepath: filename.to_string(),
+        filepath: match filename {
+            FileName::Custom(path) => format!("<{}>", path),
```
Comment (kdy1) [inline packages/next-swc/crates/next-custom-transforms/src/transforms/react_server_components.rs:646]: This is to partially revert https://github.com/swc-project/swc/pull/8530

**Review by sokra — APPROVED** [review]

**Pull request merged**